tools/swift-format/Sources/Rules/DeclSyntax+Comments.swift

```diff
@@ -38,6 +62,79 @@ extension DeclSyntax {
+
+    /// Removes the "///" from every line of comment
+    let docLineComments = comment.reversed().map { $0.dropFirst(3) }
+    return comment.isEmpty ? nil : docLineComments.joined(separator: "\n")
   }
+
+  var docCommentInfo: parseComment? {
+    guard let docComment = self.docComment else { return nil }
+    let comments = docComment.components(separatedBy: .newlines)
+    var params = [parameter]()
+    var commentParagraphs = [String]()
+    var hasFoundParameterList = false
+    var hasFoundReturn = false
+    var returnsDesc: String?
+    // Takes the first sentence of the comment, and counts the number of lines it uses.
+    let oneSenteceSummary = docComment.components(separatedBy: ".").first
+    let numOfOneSentenceLines = oneSenteceSummary!.components(separatedBy: .newlines).count
+
+    // Iterates to all the comments after the one sentence summary to find the parameter(s)
+    // return tags and get their description.
+    for line in comments.dropFirst(numOfOneSentenceLines) {
+      let trimmedLine = line.trimmingCharacters(in: .whitespaces)
+      
+      if trimmedLine.starts(with: "- Parameters") {
+        hasFoundParameterList = true
+      }
+      else if trimmedLine.starts(with: "- Parameter") {
+        // If it's only a parameter it's information is inline eith the parameter
+        // tag, just after the ':'.
+        guard let index = trimmedLine.firstIndex(of: ":") else { continue }
+        let name = trimmedLine.dropFirst("- Parameter".count)[..<index]
+          .trimmingCharacters(in: .init(charactersIn: " -:"))
+        let desc = trimmedLine[index...].trimmingCharacters(in: .init(charactersIn: " -:"))
+        params.append(parameter(paramName: name, paramDesc: desc))
+      }
+      else if trimmedLine.starts(with: "- Returns:") {
+        hasFoundParameterList = false
+        hasFoundReturn = true
+        returnsDesc = String(trimmedLine.dropFirst("- Returns:".count))
+      }
+      else if hasFoundParameterList {
+        // After the paramters tag is found the following lines should be the parameters
+        // description.
+        guard let index = trimmedLine.firstIndex(of: ":") else { continue }
+        let name = trimmedLine[..<index].trimmingCharacters(in: .init(charactersIn: " -:"))
+        let desc = trimmedLine[index...].trimmingCharacters(in: .init(charactersIn: " -:"))
+        params.append(parameter(paramName: name, paramDesc: desc))
+      }
+      else if hasFoundReturn {
+        // Appends the return description that is not inline with the return tag.
+        returnsDesc!.append(trimmedLine)
+      }
+      else if trimmedLine != ""{
+        commentParagraphs.append(" " + trimmedLine)
+      }
+    }
+    return parseComment(
+      oneSentenceSummary: oneSenteceSummary,
+      commentParagraphs: commentParagraphs,
+      parameters: params,
+      returnsDesc: returnsDesc
+    )
+  }
+}
+
+struct parameter {
+  var paramName: String
```

Comment:
Avoid abbreviations in property/function names. `name` would be fine for this property, and you could call `paramDesc` `summary` instead. (I would say `description`, but that's considered to be a fairly standard name used by the `CustomStringConvertible` protocol so it's best to avoid it.)